### PR TITLE
Create "Explore More" page and connect it to the Cards

### PR DESCRIPTION
**Pull request by Mykhailo-Chudyk**
## Description

Create an "Explore More" page and connect it to the Cards. The design for that can be found [here](https://www.figma.com/design/vnyb5qwvnjpDVaO7rOC3ZL/ID8-High-Fidelity-Designs?node-id=354-86&node-type=frame&t=thgs4GxFWZTk46M7-0). I'm currently using hardcoded data, but it should be easily replaced with the data from the backend. The buttons "Like" and "Post" are also disabled for now.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Testing Instructions

Describe the tests that you ran to verify your changes. Provide instructions so that reviewers can reproduce the results. List any relevant details for your test configuration.

## Screenshots (if applicable)

If your changes include visual components, add screenshots showing the affected pages or elements.
image
image

### CODE REVIEW

File: src/app/home/page.tsx [hunk 1]
```diff
@@ -1,8 +1,128 @@
-import React from "react";
+"use client";
```
Comment (lewiskyron) [inline src/app/home/page.tsx:1]: Given this is the home page we want to maintain the benefits of Next.js server side rendering. Now if you put "use client" at the top this file you change this file and all components imported into this page into a client side component. Now though this is a simple fix(I assume you may have put this due to running into some errors) it is not best practice. I advise we remove the "use client" and resolve what errors that probed you to change this to a client component.

File: src/app/home/page.tsx [hunk 2]
```diff
@@ -1,8 +1,128 @@
...
+import React, { useState } from "react";
 import { Box, Container } from "@mui/material";
 import Card from "../../components/Card";
+import ExploreMore from "../../components/ExploreMore";
```
Comment (lewiskyron) [inline src/app/home/page.tsx:5]: Great! Thanks for keeping our components modularized for easier managing.

File: src/app/home/page.tsx [hunk 3]
```diff
@@ -1,8 +1,128 @@
...
+import Feedback from "../../components/Feedback";
+
+interface FeedbackData {
+	name: string;
+	timeAgo: string;
+	feedback: string;
+}
+
+interface CardData {
```
Comment (lewiskyron) [inline src/app/home/page.tsx:17]: Add a comment here to probe whoever hooks this to the back-end to ensure this interface is consistent with the database tables.

File: src/app/home/page.tsx [hunk 4]
```diff
@@ -18,13 +138,11 @@ export default function Home() {
 			}}
 		>
 			<Box sx={{ position: "absolute", top: "20px", left: "20px" }}>
-				{/* todo: Insert the logo here */}
```
Comment (lewiskyron) [inline src/app/home/page.tsx]: Why was this changed. I think it may be used for those who have the todo extension where they can easily see what tasks need to be done using their vscode TODO extension. Not sure if that was the intended use case but we can maintain it as it was.

File: src/components/Card.tsx [hunk 5]
```diff
@@ -77,7 +83,6 @@ const Card = () => {
 					sx={{
 						color: "var(--color-text-secondary)",
```
Comment (lewiskyron) [inline src/components/Card.tsx:84]: good use of css variables here. Nice!

File: src/components/Card.tsx [hunk 6]
```diff
@@ -122,7 +127,7 @@ const Card = () => {
 					flexShrink: 0,
 					fontFamily: "var(--font-family-outfit)",
 				}}
-				onClick={handleExplainMore}
+				onClick={onExploreMore}
```
Comment (lewiskyron) [inline src/components/Card.tsx:130]: I cannot see where this function is defined. Not sure if you forgot to add it since the handleExplainMore was initially there but it has now been removed.

File: src/components/Card.tsx [hunk 7]
```diff
@@ -5,17 +5,23 @@ import Likes from "./Likes";
...
-		// Code to display more project information
-	};
+interface CardProps {
+	name: string;
+	date: string;
+	views: number;
+	comments: number;
+	likes: number;
+	onExploreMore: () => void;
+}
```
Comment (katyaivaniuk) [inline src/components/Card.tsx:15]: Replacing handleExplainMore with onExploreMore was a good idea, but make sure onExploreMore is fully defined in the main logic, as it looks like the handler isn’t here in the component.

File: src/components/ExploreMore.tsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
+import TitleResponse from "./TitleResponse";
+
+interface ExploreMoreData {
+	headline: string;
+	startupName: string;
+	descriptionShort: string;
+	productLink: string;
+	demoLink: string;
+	descriptionLong: string;
```
Comment (katyaivaniuk) [inline src/components/ExploreMore.tsx:13]: The layout for displaying startup details looks clear and actually very easy to read, and including clickable links for productLink and demoLink is a nice touch. Maybe just add handling for missing links to avoid broken URLs.

File: src/components/ExploreMore.tsx [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+			<Box
+				sx={{
+					width: "100%",
+					display: "flex",
+					alignItems: "center",
+					justifyContent: "center",
+					position: "relative",
+				}}
+			>
+				<ArrowBackIosNewIcon
```
Comment (katyaivaniuk) [inline src/components/ExploreMore.tsx:48]: Using ArrowBackIosNewIcon for the back button is helpful for navigation. I am thinking adding a tooltip here could be a small improvement for accessibility.

File: src/components/FeedbackItem.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+import { Box, Typography } from "@mui/material";
+
+interface FeedbackItemProps {
+	name: string;
+	timeAgo: string;
+	feedback: string;
+}
+
+const FeedbackItem: React.FC<FeedbackItemProps> = ({
+	name,
```
Comment (katyaivaniuk) [inline src/components/FeedbackItem.tsx:12]: Breaking each feedback item into its own FeedbackItem component makes the code so much cleaner, this is a great idea. I’d suggest using a unique identifier as the key instead of index if possible, this could help with rendering performance, but might be a bit of additional work.

File: src/app/home/page.tsx (hunk 3 above)
Comment (katyaivaniuk) [reply src/app/home/page.tsx:17]: I agree with the suggestion to add comments to the CardData and FeedbackData interfaces, especially so the backend team can stay in sync with any changes. This will help prevent any mismatches between the frontend and backend data.

File: src/app/home/page.tsx [hunk 11]
```diff
@@ -1,8 +1,128 @@
...
+	headline: string;
+	descriptionShort: string;
+	productLink: string;
+	demoLink: string;
+	descriptionLong: string;
+	feedbackData: FeedbackData[];
+}
 
 export default function Home() {
+	const [view, setView] = useState<"cards" | "explore">("cards");
```
Comment (katyaivaniuk) [inline src/app/home/page.tsx:33]: The state management here (especially with switching between "cards" and "explore" views) looks very clean and easy to follow. It seems flexible enough for now, though if we add more views, it might need adjusting.

**Comment by katyaivaniuk** [comment]
Great job on this PR! The structure is well-organized, and your use of modular components (like Card, ExploreMore, and Feedback) makes the codebase much more maintainable and scalable. I am pretty sure everyone will appreciate the clean handling of state transitions between "cards" and "explore" views, which keeps the user flow simple and intuitive.

A few suggestions to reiterate both of the reviews by Kyron and I:

- Consider removing "use client" in page.tsx for optimal server-side rendering on the homepage
- Adding comments to the CardData and FeedbackData interfaces would help ensure backend and frontend consistency (reduces potential mismatches.)
- For list rendering in FeedbackItem, using a unique identifier as the key rather than index would enhance React’s rendering performance.

**Review by katyaivaniuk — COMMENTED** [review]
Great job on this PR! Nice use of modular components (like Card, ExploreMore, and Feedback) definitely makes the codebase much more maintainable and scalable. 

A few suggestions to reiterate both of the reviews by Kyron and me:

- Consider removing "use client" in page.tsx for optimal server-side rendering on the homepage
- Adding comments to the CardData and FeedbackData interfaces would help ensure backend and frontend consistency (reduces potential mismatches.)
- For list rendering in FeedbackItem, using a unique identifier as the key rather than index would enhance React’s rendering performance.

**Review by katyaivaniuk — APPROVED** [review]

**Comment by avalucianelson** [comment]
This looks amazing, Mykhalio! Thank you so much for this contribution. Kyron and Katya's comments are great as well, thank you both for those. This is a really significant PR with huge implications for our MVP and product, so truly thank you for your hard work :)) 

I think the design looks perfect especially in terms of being responsive to mobile (from my best judgement). But, I think we should confirm that with @matviikotolyk since he offered to manage making all pages mobile-friendly.

**Comment by matviikotolyk** [comment]
@avalucianelson there are some visual bugs when checking the page in the mobile dimensions. In any case, I think it's easier to merge this PR once the comments are addressed and add a separate PR with the fixes across the app since most of the pages have a similar layout and there will be a lot of repetition in terms of styling.

Screenshot 2024-11-14 at 16 21 34

File: src/app/home/page.tsx (hunk 3 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx:17]: Agreed!

File: src/app/home/page.tsx (hunk 4 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx]: Great catch! I think I accidentally removed it

File: src/components/Card.tsx (hunk 6 above)
Comment (Mykhailo-Chudyk) [reply src/components/Card.tsx:130]: I'm passing this function now from the parent component
image

**Comment by Mykhailo-Chudyk** [comment]
> @avalucianelson there are some visual bugs when checking the page in the mobile dimensions. In any case, I think it's easier to merge this PR once the comments are addressed and add a separate PR with the fixes across the app since most of the pages have a similar layout and there will be a lot of repetition in terms of styling.
> 
> Screenshot 2024-11-14 at 16 21 34

Agree! We might have one person responsible for fixing all the small bugs related to responsiveness

File: src/components/FeedbackItem.tsx (hunk 10 above)
Comment (Mykhailo-Chudyk) [reply src/components/FeedbackItem.tsx:12]: That is a really great point!

File: src/components/ExploreMore.tsx (hunk 9 above)
Comment (Mykhailo-Chudyk) [reply src/components/ExploreMore.tsx:48]: Maybe we can go back to this later. There is no tooltip in the current design, so I would need to get input from Saleh on that. We can definitely go back to this later

File: src/components/ExploreMore.tsx (hunk 8 above)
Comment (Mykhailo-Chudyk) [reply src/components/ExploreMore.tsx:13]: That's a great point on the edge cases. I added a check now

File: src/components/Card.tsx (hunk 7 above)
Comment (Mykhailo-Chudyk) [reply src/components/Card.tsx:15]: Yes, it's defined in the parent component

File: src/app/home/page.tsx (hunk 1 above)
Comment (Mykhailo-Chudyk) [reply src/app/home/page.tsx:1]: Totally agree! If you don't mind, I will create a separate PR slightly refactoring this page so that it can be server rendered

**Review by katyaivaniuk — APPROVED** [review]
Looks good to me!

File: src/app/home/page.tsx (hunk 1 above)
Comment (FlaIespa) [reply src/app/home/page.tsx:1]: Understood created this issue https://github.com/ucey-star/ID8/issues/29 to keep track of this. Once you fix this in another PR feel free to close the issue. Thanks!

File: src/components/Card.tsx (hunk 6 above)
Comment (FlaIespa) [reply src/components/Card.tsx:130]: Great! Looks Good.

**Review by FlaIespa — APPROVED** [review]
